src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/BodySubstitutionParser.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection.Metadata;
+using Internal.TypeSystem;
+using System.Xml.XPath;
+using System.Globalization;
+
+namespace ILCompiler
+{
+    internal class BodySubstitutionsParser : ProcessLinkerXmlBase
+    {
+        private readonly Dictionary<MethodDesc, BodySubstitution> _methodSubstitutions;
+        private readonly Dictionary<FieldDesc, object> _fieldSubstitutions;
+
+        private BodySubstitutionsParser(TypeSystemContext context, UnmanagedMemoryStream documentStream, ManifestResource resource, ModuleDesc resourceAssembly, string xmlDocumentLocation, IReadOnlyDictionary<string, bool> featureSwitchValues)
+                : base(context, documentStream, resource, resourceAssembly, xmlDocumentLocation, featureSwitchValues)
+        {
+            _methodSubstitutions = new Dictionary<MethodDesc, BodySubstitution>();
+            _fieldSubstitutions = new Dictionary<FieldDesc, object>();
+        }
+
+        protected override void ProcessAssembly(ModuleDesc assembly, XPathNavigator nav, bool warnOnUnresolvedTypes)
+        {
+            ProcessTypes(assembly, nav, warnOnUnresolvedTypes);
+        }
+
+        // protected override TypeDesc? ProcessExportedType(ExportedType exported, ModuleDesc assembly, XPathNavigator nav) => null;
```

Comment:
```suggestion
```

Reply:
These commented out blocks match the reference source from IL Linker codebase in src\coreclr\tools\aot\ILCompiler.Compiler\Compiler\ReferenceSource. We want to keep them so that it's easier to keep the code in sync. We'll also probably want to uncomment them and make them build at some point.